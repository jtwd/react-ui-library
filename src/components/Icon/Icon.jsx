import React from 'react'
import styled from 'styled-components'
import {string, bool } from 'prop-types'
import getIcon from '../../theme/utils/utils.icons'
import getPropStylesFromTheme from '../../theme/utils/getPropStylesFromTheme'
import sizes from './Icon.sizes'

const tm = {
  sizes
}

const StyledIcon = styled.i`
  display: inline-block;
  
  > svg {
    display: block;
    width: 100%;
    height: auto;
  }
  
  ${props => !props.large && !props.small && getPropStylesFromTheme(tm, 'sizes', 'default')}

  ${props => props.small && getPropStylesFromTheme(tm, 'sizes', 'sm')}

  ${props => props.large && getPropStylesFromTheme(tm, 'sizes', 'lg')}
`

function Icon({icon, large, small}) {
  const myIcon = getIcon(icon)
  return <StyledIcon large={large} small={small}>{myIcon}</StyledIcon>
}

Icon.propTypes = {
  /** Name of icon to be displayed */
  icon: string.isRequired,
  /** Icon size */
  large: bool,
  /** Icon size */
  small: bool
}

Icon.defaultProps = {
  icon: '',
  large: false,
  small: false
}

export default Icon
