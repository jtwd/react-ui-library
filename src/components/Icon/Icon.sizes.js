import { md, lg, xl } from '../../theme/spacers'

const sizes = {
  default: `width: ${lg};`,
  sm: `width: ${md};`,
  lg: `width: ${xl};`
}

export default sizes