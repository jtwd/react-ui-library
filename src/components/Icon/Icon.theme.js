import { getTheme } from '../_theme/utils'

const spacing = getTheme('spacing')

export default {
  default: {
    uiSize: {
      default: {
        width: spacing.lg
      },
      small: {
        width: spacing.md
      },
      large: {
        width: spacing.xl
      }
    }
  }
}
