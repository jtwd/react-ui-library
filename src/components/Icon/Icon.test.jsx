import React from 'react'
import renderer from 'react-test-renderer'
import Icon from './Icon'

test('Icon renders correctly', () => {
  const component = renderer.create(<Icon icon="close" />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Icon renders large when passed the prop', () => {
  const component = renderer.create(<Icon icon="close" large />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('Icon renders small when passed the prop', () => {
  const component = renderer.create(<Icon icon="close" small />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
