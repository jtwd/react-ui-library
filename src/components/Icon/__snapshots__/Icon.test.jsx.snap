// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon renders correctly 1`] = `
<i
  className="sc-bdVaJa hujcIo"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m33.5 29.5q0 0.9-0.7 1.5l-3 3.1q-0.6 0.6-1.5 0.6t-1.5-0.6l-6.6-6.6-6.5 6.6q-0.7 0.6-1.6 0.6t-1.5-0.6l-3-3.1q-0.6-0.6-0.6-1.5t0.6-1.5l6.5-6.6-6.5-6.5q-0.6-0.7-0.6-1.6t0.6-1.5l3-3q0.6-0.6 1.5-0.6t1.6 0.6l6.5 6.6 6.6-6.6q0.6-0.6 1.5-0.6t1.5 0.6l3.1 3q0.6 0.7 0.6 1.5t-0.6 1.6l-6.6 6.5 6.6 6.6q0.6 0.6 0.6 1.5z"
      />
    </g>
  </svg>
</i>
`;

exports[`Icon renders large when passed the prop 1`] = `
<i
  className="sc-bdVaJa gUzpjX"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m33.5 29.5q0 0.9-0.7 1.5l-3 3.1q-0.6 0.6-1.5 0.6t-1.5-0.6l-6.6-6.6-6.5 6.6q-0.7 0.6-1.6 0.6t-1.5-0.6l-3-3.1q-0.6-0.6-0.6-1.5t0.6-1.5l6.5-6.6-6.5-6.5q-0.6-0.7-0.6-1.6t0.6-1.5l3-3q0.6-0.6 1.5-0.6t1.6 0.6l6.5 6.6 6.6-6.6q0.6-0.6 1.5-0.6t1.5 0.6l3.1 3q0.6 0.7 0.6 1.5t-0.6 1.6l-6.6 6.5 6.6 6.6q0.6 0.6 0.6 1.5z"
      />
    </g>
  </svg>
</i>
`;

exports[`Icon renders small when passed the prop 1`] = `
<i
  className="sc-bdVaJa xqwkO"
>
  <svg
    fill="currentColor"
    height="1em"
    preserveAspectRatio="xMidYMid meet"
    style={
      Object {
        "color": undefined,
        "verticalAlign": "middle",
      }
    }
    viewBox="0 0 40 40"
    width="1em"
  >
    <g>
      <path
        d="m33.5 29.5q0 0.9-0.7 1.5l-3 3.1q-0.6 0.6-1.5 0.6t-1.5-0.6l-6.6-6.6-6.5 6.6q-0.7 0.6-1.6 0.6t-1.5-0.6l-3-3.1q-0.6-0.6-0.6-1.5t0.6-1.5l6.5-6.6-6.5-6.5q-0.6-0.7-0.6-1.6t0.6-1.5l3-3q0.6-0.6 1.5-0.6t1.6 0.6l6.5 6.6 6.6-6.6q0.6-0.6 1.5-0.6t1.5 0.6l3.1 3q0.6 0.7 0.6 1.5t-0.6 1.6l-6.6 6.5 6.6 6.6q0.6 0.6 0.6 1.5z"
      />
    </g>
  </svg>
</i>
`;
